tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/LdmlConverter.java

```diff
@@ -137,76 +138,48 @@ private static PathMatcher supplementalMatcher(String... spec) {
-        CURRENCY_DATA(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("supplementalData", CURRENCY_DATA_PATHS, "curr", true)),
-        METADATA(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("metadata", METADATA_PATHS, "misc", false)),
-        META_ZONES(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("metaZones", METAZONE_PATHS, "misc", false)),
-        NUMBERING_SYSTEMS(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("numberingSystems", NUMBERING_SYSTEMS_PATHS, "misc", false)),
-        PLURALS(
-            SUPPLEMENTAL,
-            LdmlConverter::processPlurals),
-        PLURAL_RANGES(
-            SUPPLEMENTAL,
-            LdmlConverter::processPluralRanges),
-        WINDOWS_ZONES(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("windowsZones", WINDOWS_ZONES_PATHS, "misc", false)),
-        TRANSFORMS(
-            SUPPLEMENTAL,
-            LdmlConverter::processTransforms),
-        KEY_TYPE_DATA(
-            BCP47,
-            LdmlConverter::processKeyTypeData),
-
-        // Batching by type.
-        DTD_LDML(LDML, c -> c.processAll(LDML)),
-        DTD_SUPPLEMENTAL(SUPPLEMENTAL, c -> c.processAll(SUPPLEMENTAL)),
-        DTD_BCP47(BCP47, c -> c.processAll(BCP47));
-
-        public static final ImmutableSet<OutputType> ALL =
-            ImmutableSet.of(DTD_BCP47, DTD_SUPPLEMENTAL, DTD_LDML);
+        LOCALES(LDML),
+        BRKITR(LDML),
+        COLL(LDML),
+        RBNF(LDML),
+        DAY_PERIODS(SUPPLEMENTAL),
+        GENDER_LIST(SUPPLEMENTAL),
+        LIKELY_SUBTAGS(SUPPLEMENTAL),
+        SUPPLEMENTAL_DATA(SUPPLEMENTAL),
+        CURRENCY_DATA(SUPPLEMENTAL),
+        METADATA(SUPPLEMENTAL),
+        META_ZONES(SUPPLEMENTAL),
+        NUMBERING_SYSTEMS(SUPPLEMENTAL),
+        PLURALS(SUPPLEMENTAL),
+        PLURAL_RANGES(SUPPLEMENTAL),
+        WINDOWS_ZONES(SUPPLEMENTAL),
+        TRANSFORMS(SUPPLEMENTAL),
+        KEY_TYPE_DATA(BCP47);
+
+        public static final ImmutableSet<OutputType> ALL = ImmutableSet.copyOf(OutputType.values());
 
         private final CldrDataType type;
-        private final Consumer<LdmlConverter> converterFn;
 
-        OutputType(CldrDataType type, Consumer<LdmlConverter> converterFn) {
+        OutputType(CldrDataType type) {
             this.type = checkNotNull(type);
-            this.converterFn = checkNotNull(converterFn);
-        }
-
-        void convert(LdmlConverter converter) {
-            converterFn.accept(converter);
         }
 
         CldrDataType getCldrType() {
             return type;
         }
     }
 
+    // Map to convert the rather arbitrarily defined "output types" to the directories into which
+    // the data is written. This is only for "LDML" types since other mappers don't need to split
+    // data into multiple directories.
+    private static final ImmutableListMultimap<OutputType, IcuLocaleDir> TYPE_TO_DIR =
```

Comment:
This is so that we can decide what processing to do based on the desired "output type". Important since otherwise simple stuff like "regenerate only collation data" would take 15 minutes rather than 2 minutes.